### PR TITLE
Added support for Exclusion List for WebAuthN.

### LINKED ISSUE #1285: Webauthn support for excludeCredentials
### Description: 
Brave on iOS does not seem to support Webauthn excludeCredentials.   This will impact RelyingParties that wish to prevent a user from registering the same key multiple times.  

### Steps to Reproduce 
  1.  Using Brave on iOS go to https://webauthntest.azurewebsites.net/ and login
  2.  Click button for Create Credential
  3.  Change the default for RP Info to "This Domain" 
  4.  Select the box for "excludeCredentials"
  5.  Make sure User Info "Bob" is selected.
  6.  Click Create and tap the security key.
  7.  The credential is created.  
  8.  Click button for Create Credential again.
  9.  Assuming all the same values are still selected, click Create and tap the security key.

**Actual result:** 
The credential is created 2 times for Bob. 

**Expected result:**
The credential should not be created the second time.  An error should be returned. 

Something like this:
image

**Reproduces how often:** Easily reproduced

**Brave Version:** 
1.1.0 (19.06.21.17)

**Device details:** 
iPhone Xr (12.3.1)

### PR DESCRIPTION
**Pull request by Brandon-T**
- Added support for Exclusion List for WebAuthN.
- Fixes #1285
- Screenshot below added showing "Valid Credential found in exclusion list".

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [ ] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [ ] Issues are assigned to at least one epic.
- [ ] Issues include necessary QA labels:
  - [ ] `QA/(Yes|No)`
  - [ ] `release-notes/(include|exclude)`
  - [ ] `bug` / `enhancement`
- [ ] Necessary security reviews have taken place.
- [ ] Adequate test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable)

IMG_3196

### CODE REVIEW

File: Client/U2FExtensions.swift
Comment (kylehickinson) [inline Client/U2FExtensions.swift]: Why is this type set to `[Any]` instead of `[YKFFIDO2PublicKeyCredentialDescriptor]` (or even just letting it be inferred)?

File: Client/U2FExtensions.swift
Comment (kylehickinson) [inline Client/U2FExtensions.swift]: Can just be

```swift
credentialDescriptor.credentialType = YKFFIDO2PublicKeyCredentialType().then {
  $0.name = "public-key"
}
```

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: Client/U2FExtensions.swift
Comment (Brandon-T) [reply Client/U2FExtensions.swift]: It can't be inferred for some odd reason Xcode cannot determine what the return type of the closure is. I left it as `[Any]` because that's what the library's parameter is.. However I'll change it for this case.

File: Client/U2FExtensions.swift
Comment (Brandon-T) [reply Client/U2FExtensions.swift]: Fixed.

**Review by kylehickinson — APPROVED** [review]
Code is good, but needs @jumde's blessing before merge

**Comment by jumde** [comment]
Awesome, the code overall looks good. Doing a test run now, just to verify nothing broke.

**Review by jumde — APPROVED** [review]

**Comment by iccub** [comment]
Rebase to pass CI and should be ready to merge

**Pull request merged**
